contributors/devel/development.md

```diff
@@ -279,29 +279,59 @@ To build binaries for all platforms:
 make cross
 ```
 
+#### Install etcd
+
+```sh
+hack/install-etcd.sh
+```
+
 #### Test
 
 ```sh
 cd $working_dir/kubernetes
 
+# Run all hack/update-*.sh scripts
+# For small change, run "make verify" and then relevant scripts, for example:
+# hack/update-gofmt.sh (when adding new files)
+# hack/update-bazel.sh (when updating dependencies / imports)
+make update
+
 # Run every unit test
 make test
 
 # Run package tests verbosely
 make test WHAT=pkg/util/cache GOFLAGS=-v
 
 # Run integration tests, requires etcd
+# Refer [this](https://github.com/kubernetes/community/blob/master/contributors/devel/testing.md) for more info
 make test-integration
 
-# Run e2e tests
+# Run e2e tests by building test binaries, turn up a test cluster, run all tests, and tear the cluster down
+# Equivalent to: go run hack/e2e.go -- -v --build --up --test --down
+# Note: this takes long time! please run only affected tests by [specifying related flags](https://github.com/kubernetes/community/blob/master/contributors/devel/e2e-tests.md#building-kubernetes-and-running-the-tests)
 make test-e2e
+
+# (Optional) Check which e2e binaries are built
+go run hack/e2e.go -- -v --build
 ```
 
 See the [testing guide](testing.md) and [end-to-end tests](e2e-tests.md)
 for additional information and scenarios.
 
 Run `make help` for additional information on these make targets.
 
+#### Run e2e tests using local cluster
+
+You can run the e2e tests using local cluster for faster testing.
+
+```sh
+# Set up local cluster
+hack/local-up-cluster.sh
+
+# Build specific e2e test binary
+make all WHAT=test/e2e/e2e.test
+```
```

Comment:
This only shows how to build a binary, but we still need to tell users how to run e2e tests locally. Just move this part out to a follow up PR. I'd like to merge this PR first.